cmd/keymasterd/dependency_monitor.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"crypto/x509"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/Symantec/keymaster/lib/authutil"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	dependencyLatency = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "keymaster_dependency_check_duration_seconds",
+			Help:       "Dependency latency.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"type", "name", "target"},
+	)
+	dependencyLastSuccessSecondsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "keymaster_dependency_durations_since_last_success_seconds",
+			Help: "Seconds since last update",
+		},
+		[]string{"type", "name"},
+	)
+	lastSuccessLDAPPasswordTime time.Time
+	lastSuccessLDAPUserInfoTime time.Time
+)
+
+const timeoutSecs = 5
+
+func init() {
+	prometheus.MustRegister(dependencyLatency)
+	prometheus.MustRegister(dependencyLastSuccessSecondsGauge)
+}
+
+func checkLDAPURLs(ldapURLs string, name string, rootCAs *x509.CertPool) error {
```

Comment:
There should probably be a timeout for the whole function. If you have a lot of URLs to try and each has a 5 second timeout, you probably want to declare failure before they all time out.

Reply:
usually there are Not a lot of urls, and it is fine if they reach the timeout